pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -667,18 +682,19 @@ private static void sendStartupPacket(PGStream pgStream, List<StartupParam> para
 
     // Send the startup message.
     pgStream.sendInteger4(length);
-    pgStream.sendInteger2(3); // protocol major
-    pgStream.sendInteger2(0); // protocol minor
+    pgStream.sendInteger2(protocolMajor); // protocol major
+    pgStream.sendInteger2(protocolMinor); // protocol minor
     for (byte[] encodedParam : encodedParams) {
       pgStream.send(encodedParam);
       pgStream.sendChar(0);
     }
 
     pgStream.sendChar(0);
+    pgStream.setProtocolVersion(ProtocolVersion.fromMajorMinor(protocolMajor, protocolMinor));
     pgStream.flush();
   }
 
-  private static void doAuthentication(PGStream pgStream, String host, String user, Properties info) throws IOException, SQLException {
+  private static int doAuthentication(PGStream pgStream, String host, String user, Properties info) throws IOException, SQLException {
```

Comment:
The result of `doAuthentication` does not seem to be used. WDYT of reverting the method to `void`?